src/components/wallet/methods/trustAsset.ts

```diff
@@ -45,27 +45,30 @@ async function getAssetAndIssuer(wallet: Wallet) {
     homeDomain[homeDomain.length - 1] !== '/'
       ? homeDomain
       : homeDomain.slice(0, -1)
+
+  let homeDomainURL
   try {
-    new URL(homeDomain)
+    homeDomainURL = new URL(homeDomain)
   } catch (e) {
     throw 'anchor home domain is not a valid URL using HTTPS'
   }
 
   // if the issuer was not provided, extract if from the home domain's TOML
   if (!issuer && homeDomain) {
-    let resp = await fetch(homeDomain + '/.well-known/stellar.toml')
-    let tomlContents = parse(await resp.text())
-    if (!tomlContents.CURRENCIES) {
+    let toml = await StellarTomlResolver.resolve(homeDomainURL.host)
+    if (!toml.CURRENCIES) {
       throw "the home domain specified does not have a CURRENCIES section on it's TOML file"
     }
-    for (let c of tomlContents.CURRENCIES) {
+    for (let c of toml.CURRENCIES) {
       if (c.code === asset) {
         issuer = c.issuer
         break
       }
     }
     if (!issuer)
       throw `unable to find the ${asset} issuer on the home domain's TOML file`
+    // update here because homeDomain and toml are not fetched during updateAccount()
+    wallet.assets.set({ code: asset, issuer: issuer }, { homeDomain, toml })
```

Comment:
Cant deposit asset in fresh account

### Reproduce Bug

1. Create Account
2. Trust Asset
3. code = `SRT` Homedomain = `https://testanchor.stellar.org`
4. Deposit

### Response

`"Cannot read property 'homeDomain' of undefined"`

Reply:
I think this is a symptom of the first issue